bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/BulkPageTagger.java

```
@@ -0,0 +1,254 @@
+import java.util.LinkedHashMap;
+import java.util.Iterator;
+import java.util.HashSet;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * The type Bulk page tagger.
+ */
+public class BulkPageTagger extends ProcessDefinition implements Serializable {
+    private static final Logger log = LoggerFactory.getLogger(BulkPageTagger.class);
+    private static final long serialVersionUID = 798823856839772874L;
+
+    /**
+     * The constant NAME.
+     */
+    public static final String NAME = "Bulk Page Tagger";
+
+    /**
+     * The Excel file.
+     */
+    @FormField(
+            name = "Excel File",
+            description = "Provide the .xlsx file that defines the content pages and the corresponding cq:tags to be added on the pages",
+            component = FileUploadComponent.class,
+            options = {"mimeTypes=application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", "required"}
+    )
+    public transient InputStream excelFile = null;
+
+
+    @Override
+    public void init() throws RepositoryException {
+        // Nothing to be done here.
+
+    }
+
+
+    @Override
+    public void buildProcess(ProcessInstance instance, ResourceResolver rr) throws LoginException {
+        report.setName(instance.getName());
+        instance.getInfo().setDescription("Bulk Tag AEM content Pages");
+        instance.defineCriticalAction("Parse Excel File", rr, this::parseExcel);
+        instance.defineCriticalAction("Add Tags to Content Pages", rr, this::tagPages);
+
+    }
+
+
+    /**
+     * The Page tag mapping dataStructure.
+     */
+    transient volatile HashMap<String, String> pageTagMapping = new LinkedHashMap<>();
+
+    /**
+     * Parse input excel.
+     *
+     * @param manager the manager
+     * @throws Exception the exception
+     */
+    @SuppressWarnings({"squid:S3776", "squid:S1141"})
```

please add  @SuppressWarnings("squid:S112") to suppress the "Define and throw a dedicated exception instead of using a generic one" Sonar warning.